mdata/cache/ccache.go

```diff
@@ -70,46 +82,112 @@ func (c *CCache) evictLoop() {
 	}
 }
 
-// adds the given chunk to the cache, but only if the metric is sufficiently hot
-func (c *CCache) CacheIfHot(metric string, prev uint32, itergen chunk.IterGen) {
-	c.RLock()
+func (c *CCache) addLoop() {
+	bufferSize := 100
+	buffer := make([]AddTarget, bufferSize)
```

Comment:
don't you mean `make([]AddTarget, 0, bufferSize)` otherwise there will be bufferSize empty elements